RULES/libsupcxx/libsupcxx.go

```diff
@@ -1,78 +1,98 @@
 package libsupcxx
 
 import (
+	"fmt"
+	"os"
+
 	"dbt-rules/RULES/cc"
 	"dbt-rules/RULES/core"
-	"fmt"
-	"libsupcxx/RULES/boot"
-	"libsupcxx/RULES/config"
-	"libsupcxx/RULES/libsupcxx/lib"
-	linux "libsupcxx/boot/linux-x86_64"
-	"libsupcxx/boot/raspi3"
-	"libsupcxx/boot/x86_64"
+
+	"libsupcxx/RULES/libsupcxx/xlib"
+	"libsupcxx/RULES/target"
 	"libsupcxx/libruncxx"
 	libsupcxx "libsupcxx/libsupcxx/src"
 )
 
-type Dep = cc.Dep
-type Library = lib.Library
+// XLibrary and XBinary are cross-compilation rules. Each XLibrary / XBinary instance defines rules
+// for multiple targets (given in the Targets field, defaults to all available targets).
+// When the XLibrary / XBinary is built, it is compiled for all its targets (internally, each
+// goes inside its own subdirectory). Other rules can access a specific XLibrary / XBinary build
+// with the ForTarget(target.Target) function. XLibraries / XBinares can depend on XLibraries,
+// so that the dependency is built for the same target as the XLibrary / XBinary that uses it.
+//
+// XBinary describes a full executable, so it is also linked with the boot libraries
+
+type XLibrary = xlib.XLibrary
```

Comment:
Why alias?

Reply:
To give users outside libsupcxx a single point of access to both XLibrary and XBinary. Like cc.Library and cc.Binary. We do this in baremetal too. I would have preferred defining XLibrary and XBinary in the same package in the first place, but I had to split them to avoid a circular dependency, but that is not an issue for code outside libsupcxx.